src/contracts/actions.ts

```diff
@@ -27,12 +28,10 @@ export const overrideContractActions = (client: GenLayerClient<Transport, Simula
     account: Account;
     address: Address;
     functionName: string;
-    args: any[];
+    args: string[];
   }): Promise<any> => {
     const {account, address, functionName, args: params} = args;
-    const methodParamsAsString = JSON.stringify(params);
-    const data = [functionName, methodParamsAsString];
-    const encodedData = toRlp(data.map(param => toHex(param)));
+    const encodedData = encode({method: functionName, args: params.map(parse)});
```

Comment:
Then here you can change parse to encode, and everything will work like charm

Reply:
In the frontend I parse, because input comes from a human user typing letters in input boxes, and not from a js code